### PR TITLE
outbound messages parser added

### PR DESCRIPTION
**Pull request by logoutdhaval**
## Description

* Adding oubound message parser for message gateway.

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: src/main/java/hu/dpc/phee/operator/config/BpmnProcess.java [hunk 1]
```diff
@@ -34,4 +34,13 @@ public String getType() {
     public void setType(String type) {
         this.type = type;
     }
+
+    @Override
+    public String toString() {
```
Comment (fynmanoj) [inline src/main/java/hu/dpc/phee/operator/config/BpmnProcess.java:39]: why is this needed?

File: src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    public void setResponse(String response) {
+        this.response = response;
+    }
+
+    @Column(name = "response")
+    private String response;
+
+    @Override
+    public String toString() {
+        return "OutboudMessages{" +
```
Comment (fynmanoj) [inline src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java:159]: not needed

File: src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboundMessagesRepository.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package hu.dpc.phee.operator.entity.outboundmessages;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+public interface OutboundMessagesRepository extends JpaRepository<OutboudMessages, Long>, JpaSpecificationExecutor {
+    OutboudMessages findByInternalId(Long internalId);
```
Comment (fynmanoj) [inline src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboundMessagesRepository.java:6]: use Optional< OutboudMessages> type here

File: src/main/java/hu/dpc/phee/operator/importer/VariableParser.java [hunk 4]
```diff
@@ -90,6 +94,16 @@ public VariableParser() {
         batchParsers.put("completedAmount", pair -> pair.getFirst().setCompletedAmount(Long.parseLong(pair.getSecond())));
         batchParsers.put("resultFile", pair -> pair.getFirst().setResult_file(strip(pair.getSecond())));
         batchParsers.put("paymentMode", pair -> pair.getFirst().setPaymentMode(strip(pair.getSecond())));
+
+        outboundMessageParsers.put("tenantId",pair -> pair.getFirst().setTenantId(Long.parseLong(pair.getSecond())));
```
Comment (fynmanoj) [inline src/main/java/hu/dpc/phee/operator/importer/VariableParser.java:98]: check for possible null references

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -46,6 +46,7 @@ bpmn:
   transfer-type: "TRANSFER"
   transaction-request-type: "TRANSACTION-REQUEST"
   batch-type: "BATCH"
+  outbound-message-type: "M_OUTBOUND_MESSAGES"
```
Comment (fynmanoj) [inline src/main/resources/application.yml:49]: why does this name is same as the table name?

File: src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+import hu.dpc.phee.operator.entity.parent.AbstractPersistableCustom;
+
+import javax.persistence.*;
+
+import hu.dpc.phee.operator.entity.transfer.TransferStatus;
+import hu.dpc.phee.operator.util.SmsMessageStatusType;
+import java.util.Date;
+
+@Entity
+@Table(name = "m_outbound_messages")
```
Comment (fynmanoj) [inline src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java:12]: where is the migration script for this table? are we pushing this info to the message gateway DB or the tenant DB?

could not find the DB connection details to the message gateway DB

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java (hunk 6 above)
Comment (fynmanoj) [reply src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java:12]: what is the resolution ?

File: src/main/java/hu/dpc/phee/operator/importer/VariableParser.java (hunk 4 above)
Comment (fynmanoj) [reply src/main/java/hu/dpc/phee/operator/importer/VariableParser.java:98]: what is the resolution??

**Review by fynmanoj — CHANGES_REQUESTED** [review]
what is the resolution?

File: src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java (hunk 6 above)
Comment (fynmanoj) [reply src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java:12]: we pushing this info to the message gateway DB? or tenant db?

Comment (logoutdhaval) [reply src/main/java/hu/dpc/phee/operator/entity/outboundmessages/OutboudMessages.java:12]: message gateway db

**Comment by fynmanoj** [comment]
Describe the changes made and why they were made.
Add a link to teh design document or include the design bullet points related to this PR here.
(Ignore if these details are present on the associated JIRA ticket)

**Comment by fynmanoj** [comment]
resolve conflicts

**Comment by logoutdhaval** [comment]
> Describe the changes made and why they were made. Add a link to teh design document or include the design bullet points related to this PR here. (Ignore if these details are present on the associated JIRA ticket)

https://docs.google.com/document/d/1_oE3DvfHU_XkEx4Vd0LoL_qxVqh0eyrMSbYC9bZhCQI/edit
